### PR TITLE
Flex: implement Sandpack code editor examples

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Flex: implement Sandpack code editor examples

### CODE REVIEW

File: docs/examples/flex/flexItem.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// @flow strict
+import { type Node } from 'react';
+import { Flex, Box, Button } from 'gestalt';
+
+export default function Example(): Node {
+  return (
+    <Flex width="100%" height="100%" justifyContent="center" alignItems="center">
+      <Box borderStyle="sm" paddingX={2} paddingY={3} rounding={3} width="100%">
```
Comment (ayeshakmaz) [inline docs/examples/flex/flexItem.js:8]: The layout on this one is a bit funky now, maybe remove the border on this Box

File: docs/examples/flex/overflowing.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// @flow strict
+import { type Node } from 'react';
+import { Flex, Box, Text, Icon, Badge } from 'gestalt';
+
+export default function Example(): Node {
+  return (
+    <Flex width="100%" height="100%" justifyContent="center" alignItems="center">
+      <Box borderStyle="sm" padding={3} rounding={3}>
```
Comment (ayeshakmaz) [inline docs/examples/flex/overflowing.js:8]: Same here, can remove the border

File: docs/examples/flex/overflowing.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        <Flex alignItems="center" gap={3}>
+          <Icon accessibilityLabel="Private" icon="lock" />
+
+          <Flex.Item minWidth={0}>
+            <Text>
```
Comment (ayeshakmaz) [inline docs/examples/flex/overflowing.js:13]: We mention truncation in the description of this section. Should the Text have lineClamp={1} to show that? I don't see any difference in this example when the minWidth={0} is removed. 

Doesn't have to be fixed in this PR, just thought I'd bring it up while we're here

**Review by ayeshakmaz — APPROVED** [review]
Some comments, but looks good otherwise!

File: docs/examples/flex/flexItem.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/examples/flex/flexItem.js:8]: DONE

File: docs/examples/flex/overflowing.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/examples/flex/overflowing.js]: DONE

File: docs/examples/flex/overflowing.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/examples/flex/overflowing.js:8]: DONE

**Pull request merged**